core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1446,28 +1449,30 @@ public void run() {
     @Override
     public ManagedChannelBuilder<?> createResolvingOobChannelBuilder(String target) {
       final class ResolvingOobChannelBuilder
-          extends AbstractManagedChannelImplBuilder<ResolvingOobChannelBuilder> {
-        int defaultPort = -1;
+          extends ForwardingChannelBuilder<ResolvingOobChannelBuilder> {
+        private final ManagedChannelImplBuilder managedChannelImplBuilder;
 
         ResolvingOobChannelBuilder(String target) {
-          super(target);
+          final class ResolvingOobUnsupportedClientTransportFactoryBuilder extends
+              UnsupportedClientTransportFactoryBuilder {}
+
+          managedChannelImplBuilder = new ManagedChannelImplBuilder(target,
+              new ResolvingOobUnsupportedClientTransportFactoryBuilder(),
```

Comment:
Use UnsupportedClientTransportFactoryBuilder directly?

Reply:
The idea was to make it explicit in stack traces. Is it unnecessary?

Reply:
Since it should never be used, that seems of limited value, also because it seems it would be easy to debug as the OOB channel is used in very different circumstances.